Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -95,49 +95,73 @@ float ScrollSnapAnimatorState::adjustedScrollDestination(ScrollEventAxis axis, F
     return offset * pageScale;
 }
 
-void ScrollSnapAnimatorState::setFocusedElementForAxis(ScrollEventAxis axis)
+bool ScrollSnapAnimatorState::preserveCurrentTargetForAxis(ScrollEventAxis axis, uint64_t box)
 {
     auto snapOffsets = snapOffsetsForAxis(axis);
-    auto found = std::find_if(snapOffsets.begin(), snapOffsets.end(), [](SnapOffset<LayoutUnit> p) -> bool {
-        return p.isFocused;
-    });
-    if (found == snapOffsets.end())
-        return;
-
-    auto newIndex = std::distance(snapOffsets.begin(), found);
-    auto newID = snapOffsets[newIndex].snapTargetID;
-    setActiveSnapIndexForAxis(axis, newIndex);
-    setActiveSnapIndexIDForAxis(axis, newID);
-}
-
-bool ScrollSnapAnimatorState::preserveCurrentTargetForAxis(ScrollEventAxis axis)
-{
-    auto snapID = activeSnapIDForAxis(axis);
-    auto snapOffsets = snapOffsetsForAxis(axis);
     
-    auto found = std::find_if(snapOffsets.begin(), snapOffsets.end(), [snapID](SnapOffset<LayoutUnit> p) -> bool {
-        return p.snapTargetID == *snapID;
+    auto found = std::find_if(snapOffsets.begin(), snapOffsets.end(), [box](SnapOffset<LayoutUnit> p) -> bool {
+        return p.snapTargetID == box;
     });
     if (found == snapOffsets.end()) {
-        setActiveSnapIndexIDForAxis(axis, std::nullopt);
+        setActiveSnapIndexForAxis(axis, std::nullopt);
         return false;
     }
     
     setActiveSnapIndexForAxis(axis, std::distance(snapOffsets.begin(), found));
     return true;
 }
 
-bool ScrollSnapAnimatorState::resnapAfterLayout(ScrollOffset scrollOffset, const ScrollExtents& scrollExtents, float pageScale)
+Vector<SnapOffset<LayoutUnit>> ScrollSnapAnimatorState::currentlySnappedOffsets(ScrollEventAxis axis)
 {
-    // Check if we need to set the active target to a focused element
-    setFocusedElementForAxis(ScrollEventAxis::Vertical);
-    setFocusedElementForAxis(ScrollEventAxis::Horizontal);
+    Vector<SnapOffset<LayoutUnit>> currentlySnappedOffsets;
+    auto snapOffsets = snapOffsetsForAxis(axis);
+    auto activeIndex = activeSnapIndexForAxis(axis);
     
+    if (activeIndex && *activeIndex < snapOffsets.size())
+        currentlySnappedOffsets.append(snapOffsets[*activeIndex]);
+    return currentlySnappedOffsets;
+}
+
+static bool isSnappedToMultipleBoxes(Vector<SnapOffset<LayoutUnit>> verticalOffsets, Vector<SnapOffset<LayoutUnit>> horizontalOffsets)
+{
+    HashSet<uint64_t> boxIDs;
+    verticalOffsets.appendVector(horizontalOffsets);
```

Comment:
This is confusing; it's OK to append the vectors because all you care about is the snapTargetIDs? Seems like there would be a better way to write this code.

Reply:
It's a bit tricky since we are tracking the offsets per axis but we want to perform an operation on all of them. I have changed this and the other sights to explicitly make a separate vector which should hopefully make this clearer.